drivers/serial/uart_gecko.c

```diff
@@ -354,19 +498,77 @@ static const struct uart_driver_api uart_gecko_driver_api = {
 			gpioModePushPull, 1				       \
 	}
 
-#define GECKO_UART_PINS(idx)						       \
+#define GECKO_UART_RX_TX_PINS(idx)					       \
 	.pin_rx = PIN_UART_RXD(idx),					       \
 	.pin_tx = PIN_UART_TXD(idx),
 
+#ifdef UART_GECKO_HW_FLOW_CONTROL
+
+#ifdef CONFIG_SOC_GECKO_HAS_INDIVIDUAL_PIN_LOCATION
+#define GECKO_UART_RTS_CTS_PIN_LOCATIONS(idx)				       \
+	.loc_rts = COND_CODE_1(DT_INST_NODE_HAS_PROP(idx, location_rts),       \
+			       (DT_INST_PROP_BY_IDX(idx, location_rts, 0)),    \
+			       (0)),					       \
+	.loc_cts = COND_CODE_1(DT_INST_NODE_HAS_PROP(idx, location_rts),       \
+			       (DT_INST_PROP_BY_IDX(idx, location_cts, 0)),    \
+			       (0)),
+#define VALIDATE_GECKO_UART_RTS_CTS_PIN_LOCATIONS(idx)
+#else /* CONFIG_SOC_GECKO_HAS_INDIVIDUAL_PIN_LOCATION */
+#define GECKO_UART_RTS_CTS_PIN_LOCATIONS(idx)
+#define VALIDATE_GECKO_UART_RTS_CTS_PIN_LOCATIONS(idx)			       \
+	BUILD_ASSERT(DT_INST_PROP_BY_IDX(idx, location_rts, 0) ==	       \
+		      DT_INST_PROP_BY_IDX(idx, location_cts, 0),	       \
+		     "DTS location-* properties must have identical value")
+
+#endif /* CONFIG_SOC_GECKO_HAS_INDIVIDUAL_PIN_LOCATION */
+
+#define PIN_UART_RTS(idx)						       \
+	COND_CODE_1(DT_INST_NODE_HAS_PROP(idx, location_rts),		       \
+	({								       \
+		DT_INST_PROP_BY_IDX(idx, location_rts, 1),		       \
+			DT_INST_PROP_BY_IDX(idx, location_rts, 2),	       \
+			gpioModePushPull, 1				       \
+	}),								       \
+	({}))
```

Comment:
I believe we can't do this. The structure will be initialized to all 0s. The side effect of running `soc_gpio_configure` with such data will be to disable pin PA0. The code was probably supposed not to configure RTS pin at all but that's not the case. It may be easier to force user to provide location property for RTS, CTS pins. Otherwise implementation becomes complicated.

Reply:
I had to leave this COND_CODE_1 macros to be able to compile the driver  when one post has enabled hw flow control, and one other port is used without hw flow control.

The rts and cts pins are only configured, if hw flow control is enabled:
```
if (config->hw_flowcontrol) {
		soc_gpio_configure(&config->pin_rts);
		soc_gpio_configure(&config->pin_cts);
```

Reply:
My bad, I overlooked this. Thanks for the comment.